chore: enhance .NET onboarding story (#15755)
microsoft/playwright: docs/src/writing-tests-csharp.md

```

### Test Isolation

The Playwright NUnit and MSTest test framework base classes will isolate each test from each other by providing a separate `Page` instance. Pages are isolated between tests due to the Browser Context, which is equivalent to a brand new browser profile, where every test gets a fresh environment, even when multiple tests run in a single Browser.

<Tabs
  defaultValue="nunit"
  values={[
    {label: 'NUnit', value: 'nunit'},
    {label: 'MSTest', value: 'mstest'}
  ]
}>
<TabItem value="nunit">

```csharp
using System.Threading.Tasks;
using Microsoft.Playwright.NUnit;
using NUnit.Framework;

namespace PlaywrightTests;

[Parallelizable(ParallelScope.Self)]
public class Tests : PageTest
{
    [Test]
    public async Task BasicTest()
    {
        await Page.GotoAsync("https://playwright.dev");
    }
}
```

</TabItem>
<TabItem value="mstest">

```csharp
using Microsoft.Playwright.MSTest;

namespace PlaywrightTests;

public class UnitTest1 : PageTest
{
    [TestMethod]
    public async Task BasicTest()
    {
        await Page.GotoAsync("https://playwright.dev");
    }
}
```

</TabItem>
</Tabs>

### Using Test Hooks

> You can use `SetUp` and `TearDown` (or `TestInitialize` and `TestCleanup` in MSTest) as usual to prepare and clean up your test environment:

Reviewer: ```suggestion
You can use `SetUp`/`TearDown` in NUnit and `TestInitialize`/`TestCleanup` in MSTest to prepare and clean up your test environment:
```
